docs/mmds/mmds-user-guide.md

```diff
@@ -0,0 +1,312 @@
+                       "network": {
+                            "interfaces": {
+                                 "macs": {
+                                      "02:29:96:8f:6a:2d": {
+                                           "device-number": "13345342",
+                                           "local-hostname": "localhost",
+                                           "subnet-id": "subnet-be9b61d"
+                                      }
+                                 }
+                            }
+                       }
+                  }
+            }
+    }"
+```
+
+To partially update existing metadata, an HTTP `PATCH` request to the `/mmds`
+resource has to be issued, using as a payload the metadata patch, as
+[JSON Merge Patch](https://tools.ietf.org/html/rfc7396) functionality
+describes. A complete description of updating metadata Firecracker API can be
+found in the [firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/mmds"          \
+    -H "Content-Type: application/json"       \
+    -d "{
+            "latest": {
+                  "meta-data": {
+                       "ami-id": "ami-87654321",
+                       "reservation-id": "r-79054aef",
+                  }
+            }
+    }"
+```
+
+# Retrieving metadata
+
+MicroVM metadata can be retrieved both from host and guest operating systems.
+For the scope of this chapter, let's assume the data store content is the JSON
+below:
+
+```json
+{
+    "latest": {
+          "meta-data": {
+               "ami-id": "ami-87654321",
+               "reservation-id": "r-79054aef"
+          }
+    }
+}
+```
+
+## Retrieving metadata in the host operating system
+
+To retrieve existing MMDS metadata from host operating system, an HTTP `GET`
+request to the `/mmds` resource must be issued. The HTTP response returns the
+existing metadata, as a JSON formatted text. A complete description of
+retrieving metadata Firecracker API can be found in the
+[firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+curl -s --unix-socket /tmp/firecracker.socket http://localhost/mmds
+```
+
+Output:
+
+```json
+{
+    "latest": {
+          "meta-data": {
+               "ami-id": "ami-87654321",
+               "reservation-id": "r-79054aef"
+          }
+    }
+}
+```
+
+## Retrieving metadata in the guest operating system
+
+To retrieve existing MMDS metadata from guest operating system, an HTTP `GET`
+request must be issued. The requested resource can be referenced by its
+corresponding [JSON Pointer](https://tools.ietf.org/html/rfc6901), which is
+also the path of the MMDS request. The HTTP response content will contain the
+referenced metadata resource. The response format can be JSON (experimental) or
+IMDS. The IMDS documentation can be found
```

Comment:
Add something like, one can select the output format by specifying the optional accept header, application/json will format the output to json, while using plain-text or not specifying this optional header will format the output to imds